r55/src/exec.rs

```diff
@@ -95,10 +118,27 @@ fn riscv_context(frame: &Frame) -> Option<RVEmu> {
     let interpreter = frame.interpreter();
 
     let Some((0xFF, bytecode)) = interpreter.bytecode.split_first() else {
+        warn!("NOT RISCV CONTRACT!");
         return None;
     };
 
-    match setup_from_elf(bytecode, &interpreter.contract.input) {
+    let (code, calldata) = if frame.is_create() {
+        let (code_size, init_code) = bytecode.split_at(4);
+        let Some((0xFF, bytecode)) = init_code.split_first() else {
+            warn!("NOT RISCV CONTRACT!");
+            return None;
+        };
+        let code_size = U32::from_be_slice(code_size).to::<usize>() - 1; // deduct control byte `0xFF`
+        let end_of_args = init_code.len() - 34; // deduct control byte + ignore empty (32 byte) word appended by revm
```

Comment:
revm seems to add an 32 empty bytes at the end of the constructor bytecode  >>  i used the initcode length as a reference and then substracted the diff to properly extract the encoded args.
instead, would we prefer to change the initcode format?
   - `[0xFF][codesize][bytecode][constructor_args]` (current)
   - `[0xFF][codesize][constructor_args_size][bytecode][constructor_args]` (proposal)
  
the latter looks cleaner to me, but perhaps unnecessary idk... thoughts? @leonardoalt @gakonst